Improve privacy banner

delnyn opened the pull request:
Hey this is still a WIP

Commit:
New buttons

Commit:
Update style.scss

Commit:
Update style.scss

Commit:
Update style.scss

ghost commented:
hey @locness3 could you review this?

moeenio commented:
Ofc

moeenio commented on _layouts/default.html:
```diff
@@ -36,8 +36,9 @@
     <div>
       <h2>Privacy information</h2>
       <p>We use Matomo hosted on a little Raspberry Pi for analytics.
```
I would change that text too, emphasizing that these are managed ourselves

moeenio replied:
we manage our own analytics with matomo, or smth like this

moeenio committed:
:bug: Buttons inherit font family

moeenio reviewed (commented):
I would align the buttons to the right on desktop.
![image](https://user-images.githubusercontent.com/37651007/104117074-7612cd00-531e-11eb-8e8d-c0024678054b.png)

And center everything on mobile
![image](https://user-images.githubusercontent.com/37651007/104117047-4b287900-531e-11eb-8a10-c7ea03396296.png)

moeenio commented:
Also, clicking "reject" opens up opt-out but doesn't close the banner (which require clicking "accept")

ghost commented:
OK! I'll change that

Commit:
New text

Commit:
Close Privacy Banner when "Reject" is clicked

ghost commented:
Nice it works

moeenio commented on _layouts/default.html:
```diff
@@ -37,7 +37,7 @@
       <p>We are managing our own analytics using Matomo.
       <a href="https://matomo.org/faq/general/faq_18254/" target="_blank" rel="noopener">Know what it collects</a>.</p>
-      <a href="javascript:matomoOptOut()" class="btn">Reject</a>
+      <button onclick="matomoOptOut();closePrivacyBanner()" class="btn">Reject</button>
```
why not, but that's inconsistent with the accept button

moeenio replied:
Might want to use the `js-privacy-banner-close` class on the two, and adapt the js (use querySelector**All** with a loop)

moeenio replied:
And add a `js-matomo-opt-out` class too

moeenio replied:
Instead of using onClick which is some IE9 level stuff you use the `js-privacy-banner-close` on the two, iterating through the two buttons with querySelectorAll (querySelector returns a single node while querySelectorAll returns a list of nodes)

moeenio replied:
There's no js files, only script tags.

ghost commented:
ok will do later

Commit:
Use querySelectors instead of onClick property

ghost commented:
There you go! Need to style it now

ghost commented:
- [ ] Aligns button right on desktop
- [ ] Aligns buttons center on mobile
- [x] Change the text
- [x] Use consistent tags on buttons
- [x] Use classes and querySelectors instead of onClicks

moeenio commented:
Did you test it? Because AFAIK you cant `forEach` with a `querySelector`

ghost commented:
`querySelectorAll` returns a list?

ghost commented:
OK it doesn't work but I don't know why

moeenio commented:
> `querySelectorAll` returns a list?

yeh but not a standard array

moeenio committed:
fix it

moeenio committed:
use a number to remeber banner is closed
if we update the banner, we change this number so the banner shows again if previously opted-in

moeenio commented:
fixed it

ghost commented:
nice ily
tho I pushed my css test in prod

Commit:
Merge branch 'master' into improve-privacy-banner

ghost commented:
resolved conflict!

moeenio commented:
Okk

ghost commented:
so, the problem with the querySelector is fixed? we just need to change the layout?

moeenio commented:
Yes

ghost commented:
ugh the buttons don't need to be aligned after all I guess  
I'm giving up on these last steps  
can I squash and merge @locness3 ?

moeenio commented:
If you really care about the project i'll give you access to the repo.

ghost commented:
no I don't want to maintain the project I just want to finish that PR, which is now 9 months old  
so can I squash and merge now?

moeenio commented:
Since this is not gdpr-compliant and doesn't acknowledge third-party widgets hosted at Google on the site...

Yes. Who cares anymore?
